ext/src/CoupledSDEs.jl

```diff
@@ -0,0 +1,206 @@
+    else
+        return (DEFAULT_SDE_SOLVER, diffeq)
+    end
+end
+
+###########################################################################################
+# Constructor functions
+###########################################################################################
+
+function DynamicalSystemsBase.CoupledSDEs(
+        f,
+        g,
+        u0,
+        p = SciMLBase.NullParameters(),
+        noise_strength = 1.0;
+        t0 = 0.0,
+        diffeq = DEFAULT_STOCH_DIFFEQ,
+        noise_rate_prototype = nothing,
+        noise = nothing,
+        seed = UInt64(0)
+)
+    IIP = isinplace(f, 4) # from SciMLBase
+    @assert IIP==isinplace(g, 4) "f and g must both be in-place or out-of-place"
+
+    s = correct_state(Val{IIP}(), u0)
+    T = eltype(s)
+    prob = SDEProblem{IIP}(
+        f,
+        g,
+        s,
+        (T(t0), T(Inf)),
+        p;
+        noise_rate_prototype = noise_rate_prototype,
+        noise = noise,
+        seed = seed
+    )
+    return CoupledSDEs(prob, diffeq, noise_strength)
+end
+
+function DynamicalSystemsBase.CoupledSDEs(
+        prob::SDEProblem, diffeq = DEFAULT_STOCH_DIFFEQ, noise_strength = 1.0; special_kwargs...
+)
+    if haskey(special_kwargs, :diffeq)
+        throw(
+            ArgumentError(
+            "`diffeq` is given as positional argument when an ODEProblem is provided."
+        ),
+        )
+    end
+    IIP = isinplace(prob) # from SciMLBase
+    D = length(prob.u0)
+    P = typeof(prob.p)
+    if prob.tspan === (nothing, nothing)
+        # If the problem was made via MTK, it is possible to not have a default timespan.
+        U = eltype(prob.u0)
+        prob = SciMLBase.remake(prob; tspan = (U(0), U(Inf)))
+    end
+    sde_function = SDEFunction(prob.f, add_noise_strength(noise_strength, prob.g, IIP))
+    prob = SciMLBase.remake(prob; f = sde_function)
+    noise_type = find_noise_type(prob, IIP)
+
+    solver, remaining = _decompose_into_sde_solver_and_remaining(diffeq)
+    integ = __init(
+        prob,
+        solver;
+        remaining...,
+        # Integrators are used exclusively iteratively. There is no reason to save anything.
+        save_start = false,
+        save_end = false,
+        save_everystep = false,
+        # DynamicalSystems.jl operates on integrators and `step!` exclusively,
+        # so there is no reason to limit the maximum time evolution
+        maxiters = Inf
+    )
+    return CoupledSDEs{IIP, D, typeof(integ), P, true}(
+        integ, deepcopy(prob.p), noise_strength, diffeq, noise_type
+    )
+end
+
+"""
+    CoupledSDEs(ds::CoupledODEs, g, p [, σ]; kwargs...)
+
+Converts a [`CoupledODEs`
+](https://juliadynamics.github.io/DynamicalSystems.jl/stable/tutorial/#CoupledODEs)
+system into a [`CoupledSDEs`](@ref).
+"""
+function DynamicalSystemsBase.CoupledSDEs(
+        ds::CoupledODEs,
+        g,
+        p, # the parameter is likely changed as the diffusion function g is added.
```

Comment:
as far as I understand the way DifferentialEquations.jl handles `f, g` is that both take as input the same parameter container. I also don't see any advantage of separating the parameter container into two. What is your rationale for it?

Reply:
Yes, we SDEIntegrator takes on `p` for both `f` and `g`. But, I think @reykboerner is referring to something else. Once you made a `CoupledODEs` and you want to convert to a `CoupledSDEs` the parameters can be changed. Now you have to give all parameters again. @reykboerner is thinking to let the user add only the parameters related to `g` and merge them with the parameters of `f`. However, I think this will lead to issues with ModelingToolkit and when the user wants use the same parameter in `f` and `g`. I would just leave it like this and let the user regive the whole parameter container.